src/navigation-plugin.tsx

```diff
@@ -97,15 +103,23 @@ export class NavigationPlugin extends KalturaPlayer.core.BasePlugin {
   };
 
   loadMedia(): void {
-    if (!this.cuePointManager || !this.sidePanelsManager || !this.upperBarManager) {
-      this.logger.warn("kalturaCuepoints, sidePanelsManager or upperBarManager haven't registered");
+    if (!this.config.visible) {
+      this.logger.warn("visible configuration is false - not rendering the plugin.");
+      return;
+    }
+    if (!this.cuePointManager || !this.sidePanelsManager || !this.upperBarManager || !this.timelineManager) {
```

Comment:
have you checked with David or Dana the case when if `timelineManager` absent Navigation plugin should not work?